### DISCUSSION TITLE
Copilot stopped working in devcontainer suddenly

### DISCUSSION THREAD

**Discussion by perezd** [community/community#36728] [Copilot Conversations]
Copilot was recently updated to: v1.54.7077
I just recently updated VSCode to: 
```
1.72.2 (Universal)
Commit: d045a5eda657f4d7b676dedbfa7aab8207f8a075
Date: 2022-10-12T22:16:30.254Z
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin arm64 21.6.0
Sandboxed: No
```

Prior to this, I had a working Copilot instance running from within a devcontainer, and suddenly it just stopped working...I don't see anything strange in the output log either, just this:

```
[INFO] [ghostText] [2022-10-20T01:49:31.689Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-10-20T01:49:31.757Z] Cancelled after requestMultiline
[INFO] [default] [2022-10-20T01:49:31.976Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2022-10-20T01:49:33.200Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2022-10-20T01:49:35.379Z] Cancelled during debounce
[INFO] [default] [2022-10-20T01:49:35.541Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
```

If I pop out of the devcontainer into a local VSCode Copilot works as I expect...so something must have gone wrong recently. I've also obliterated the container image, signed in/out of things, and all the standard checks.

Thoughts?

**Comment by iamcjwilliams** [comment]
@perezd Thank you for your feedback, Our copilot team is investigating this

**Reply by perezd** [reply]
I dunno what happened, but it started working again after testing it out this afternoon. Thanks!

**Comment by ossanna16** [comment]
Hey there @perezd and thank you for asking a good question! I'm glad you found our community 🙂

**Comment by Druid33** [comment]
Hello.
I have the same experience. Today Github Copilot extension stop working in VSCode devcontainers, bat localy works fine. Yesterday it works fine in devcontainers too.

There is no error mesage in log. I can see extension is succesfully instaled in container, but in runtime status window of extension i see: "Not yet activated".

Versions of VSCode and Github Copilot extensions are up to date.

```
Version: 1.77.0
Commit: 7f329fe6c66b0f86ae1574c2911b681ad5a45d63
Date: 2023-03-29T10:05:35.165Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Linux x64 5.19.0-38-generic
Sandboxed: No
```

You can simulate it in this simple devcontainer:
```
{
    "name": "Copilot extension test",
    "image": "alpine",
    "extensions": [
        "GitHub.copilot"
    ]
}
```
Some outputs of running container you can see here:
![image](https://user-images.githubusercontent.com/6436809/229835077-1a0e636b-25c5-4145-8321-27cbdd2ed13f.png)

Thanks for any advice. Github Copilot is awesome assistant and after some testing a like to recommend it at work like official tool for developers.
I will try it tomorow, maybe it start working like in @perezd  case :)

**Reply by zolzaya** [reply]
I have same issue. Any news guys???

**Comment by Druid33** [comment]
I try it next day and surprise-surprise...still not working.

But i found "hotfix". It is possible force extension not to run from container, but from UI part of vscode. Add this lines to settings.json:
```
"remote.extensionKind": {
        "GitHub.copilot": ["ui"],
    },
```

**Reply by tribal2** [reply]
Thanks!! 

In case it doesn't work for someone, these lines should be added to the settings.json file in the devcontainer, not to the local settings.json. Preferences: Open Remote Settings (JSON)

**Reply by zolzaya** [reply]
Thanks it's like a charm :)

**Reply by rahulthakur319** [reply]
Works like a charm :-)

**Reply by mitkonikov** [reply]
Even after 3 years, the same issue persists and the same fix works :D
